code/modules/mob/new_player/new_player.dm

```diff
@@ -40,6 +41,35 @@
 	privacy_consent()
 	return FALSE
 
+/mob/new_player/proc/challenge_phrase()
+	if(!config.challenge_phrase) //check the config if there is in fact a challenge phrase, if not then just return true
+		return TRUE
+	establish_db_connection()
+	if(!dbcon.IsConnected())
+		return TRUE
+	var/DBQuery/query = dbcon.NewQuery("SELECT challenge FROM [format_table_name("ban")] WHERE ckey = '[src.ckey]' AND challenge=1")
+	query.Execute()
+	while(query.NextRow()) //Find a row that matches ckey where challenge is requested

```

Comment:
Isn't this going to prompt the person multiple times if they have multiple bans that were unbanned with the challenge phrase active?

Reply:
as far as I'm aware, no since there is only one column and that column and its either true or false, pressing "yes" multiple times when an admin does the unban is equivalent of me saying
```

"var/A

A = FALSE

A = FALSE

A = FALSE"

```